src/content/getusermedia/gum/js/test.js

```diff
@@ -5,98 +5,80 @@
  *  that can be found in the LICENSE file in the root of the source
  *  tree.
  */
- /* eslint-env node */
-
+/* eslint-env node */
 'use strict';
+
 // This is a basic test file for use with testling.
 // The test script language comes from tape.
-var test = require('tape');
+const test = require('tape');
+
+const webdriver = require('selenium-webdriver');
+const seleniumHelpers = require('webrtc-utilities').seleniumLib;
 
-var webdriver = require('selenium-webdriver');
-var seleniumHelpers = require('webrtc-utilities').seleniumLib;
+const path = '/src/content/getusermedia/gum/index.html';
+const url = (process.env.BASEURL ? process.env.BASEURL : ('file://' + process.cwd())) + path;
 
-test('Video width and video height are set on GUM sample', function(t) {
+test('Video width and video height are set on GUM sample', t => {
   // FIXME: use env[SELENIUM_BROWSER] instead?
-  var driver = seleniumHelpers.buildDriver();
+  const driver = seleniumHelpers.buildDriver();
 
-  driver.get((process.env.BASEURL ? process.env.BASEURL :
-      ('file://' + process.cwd())) +
-      '/src/content/getusermedia/gum/index.html')
-  .then(function() {
-    t.pass('Page loaded');
-  })
-  .then(function() {
-    return driver.wait(function() {
+  driver.get(url)
+    .then(() => t.pass('Page loaded'))
+    .then(() => driver.wait(() => {
       return driver.executeScript('return window.stream !== undefined;');
-    }, 30 * 1000);
-  })
-  // Check that there is a stream with a video track.
-  .then(function() {
-    return driver.executeScript('return stream && stream.getTracks().length');
-  })
-  .then(function(numberOfStreams) {
-    t.ok(numberOfStreams === 1, 'Stream exists and has one track');
-  })
-  // Check that there is a video element and it is displaying something.
-  .then(function() {
-    return driver.findElement(webdriver.By.id('gum-local'));
-  })
-  .then(function(videoElement) {
-    t.pass('Found video element');
-    var width = 0;
-    var height = 0;
-    return new webdriver.promise.Promise(function(resolve) {
-      videoElement.getAttribute('videoWidth').then(function(w) {
-        width = w;
-        // TODO: Figure out why videoWidth is 0 most of the time on Chrome.
-        // TODO: After above TODO is fixed, add > zero validation.
-        t.pass('Got videoWidth ' + w);
-        if (width && height) {
-          resolve([width, height]);
-        }
-      });
-      videoElement.getAttribute('videoHeight').then(function(h) {
-        height = h;
-        t.pass('Got videoHeight ' + h);
-        if (width && height) {
-          resolve([width, height]);
-        }
+    }, 30 * 1000))
+    // Check that there is a stream with a video track.
+    .then(() => driver.executeScript('return stream && stream.getTracks().length'))
+    .then(numberOfStreams => t.ok(numberOfStreams === 1, 'Stream exists and has one track'))
```

Comment:
Drive-by comment about code, unrelated to refactoring: There is no need to verify the number of tracks in a stream. If GUM() was not rejected and we have a stream, it will have a track. Let's not unittest that the browser is not buggy unless there are known issues (there aren't).

Reply:
Yes, I have some opinions about the quality of the tests in general. Some of them are not relevant and other are testing the "wrong" things. Will fix in later PRs.

Reply:
Cool